### PR TITLE
update code in text-preprocessing

### PR DESCRIPTION
**Pull request by mli**
*Description of changes:*

Simplified the vocab code. Hasn't put into d2l.py yet, will do it once fixed other chapters. 

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1847/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1847/

**Comment by mli** [comment]
Job d2l-en/PR-1847/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1847/

File: chapter_recurrent-neural-networks/text-preprocessing.md [hunk 1]
```diff
@@ -72,113 +66,88 @@ print(lines[10])
...
+        # Expand a list of list into a list if needed
+        if tokens and isinstance(tokens[0], list):
+            tokens = [token for line in tokens for token in line]
+        # Count token frequencies
+        counter = collections.Counter(tokens)
+        self.token_freqs = sorted(counter.items(), key=lambda x: x[1],
+                                  reverse=True)
+        # The list of unique tokens. Sorting it for a deterministic mapping
+        self.idx_to_token = list(sorted(set(['<unk>'] + reserved_tokens + [
+            token for token, freq in self.token_freqs if freq >= min_freq])))
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/text-preprocessing.md:120]: `token_freqs` is already sorted so we can get a deterministic mapping from it without another sorting. Besides, it's quite common to place <unk> at index 0, followed by reserved tokens such as <pad>, <cls>, ...

**Comment by mli** [comment]
Job d2l-en/PR-1847/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1847/

**Pull request merged**